src/sphinxcontrib/towncrier/_fragment_discovery.py

```diff
@@ -34,11 +37,54 @@ def _find_config_file(base: Path) -> Path:
 # pylint: disable=fixme
 # FIXME: refactor `lookup_towncrier_fragments` to drop noqas
 @lru_cache(maxsize=1, typed=True)  # noqa: WPS210
-def lookup_towncrier_fragments(  # noqa: WPS210
+def lookup_towncrier_fragments(
         working_dir: Optional[str] = None,
         config_path: Optional[str] = None,
 ) -> Set[Path]:
     """Emit RST-formatted Towncrier changelog fragment paths."""
+    tc_version = Version(version('towncrier'))
+    if tc_version >= Version('24.7.0'):
+        return _lookup_towncrier_fragments_post24_7(working_dir, config_path)
+    return _lookup_towncrier_fragments_pre24_7(working_dir, config_path)
+
+
+# used for towncrier version 24.7 and above
+def _lookup_towncrier_fragments_post24_7(  # noqa: WPS210
+    working_dir: Optional[str] = None, config_path: Optional[str] = None,
+) -> Set[Path]:
+    project_path = Path.cwd() if working_dir is None else Path(working_dir)
+
+    final_config_path = (
+        _resolve_spec_config(project_path, config_path)
+        or _find_config_file(project_path)
+    )
+
+    try:
+        towncrier_config = get_towncrier_config(
+            project_path,
+            final_config_path,
+        )
+    except KeyError as key_err:
+        # NOTE: The error is missing key 'towncrier' or similar
+        logger.warning(
+            f'Missing key {key_err!s} in file {final_config_path!s}',
+        )
+        return set()
+
+    fragment_dir = towncrier_config.directory or 'newsfragments'
```

Comment:
It occurred to me that this is probably going to yield weird behavior with an empty string in the `directory` attribute.